feat: updated ui for adding chain id screen

NidhiKJha opened the pull request:
This PR is to update the add network chain screen
## **Related issues**

Fixes: [2660](https://github.com/MetaMask/MetaMask-planning/issues/2660)

## **Manual testing steps**

1. Run extension with CHAIN_PERMISSIONS feature flag
2. Connect with Uniswap
3. Try adding new network from Uniswap
4. See the new UI for confirmations screen

## **Screenshots/Recordings**


### **Before**

https://github.com/user-attachments/assets/0bee3a7f-741a-48bf-9d98-64aa3dbef8af



### **After**


https://github.com/user-attachments/assets/e9fa3345-2a83-4afb-be12-e637dc3a6af8


## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

NidhiKJha committed:
updated ui for adding chain id screen

NidhiKJha committed:
updated props

NidhiKJha committed:
updated props

NidhiKJha committed:
updated values

NidhiKJha committed:
updated feature flag

NidhiKJha committed:
updated ethereum chain

NidhiKJha committed:
lint fix

matthewwalsh0 commented on ui/components/ui/truncated-definition-list/truncated-definition-list.js:
```diff
@@ -18,7 +18,39 @@ export default function TruncatedDefinitionList({
   const t = useI18nContext();
 
-  return (
+  return process.env.CHAIN_PERMISSIONS ? (
```
Rather than branching the entire render, could we instead only branch when the output changes?

The top level `Box` for example is identical as are other nested elements.

NidhiKJha replied:
Created a separate function to handle this and avoid repetition here https://github.com/MetaMask/metamask-extension/pull/25777/commits/e87feb11f778de6e03ad95819414cbb2233c4b80

matthewwalsh0 commented on ui/pages/confirmations/confirmation/confirmation.js:
```diff
@@ -507,8 +506,8 @@ export default function ConfirmationPage({
-      <div className="confirmation-page__content">
-        {templatedValues.networkDisplay ? (
+      <Box className="confirmation-page__content" paddingTop={4}>
+        {templatedValues.networkDisplay && !process.env.CHAIN_PERMISSIONS ? (
```
Rather than hardcoding this here, could we instead just set `networkDisplay` based on the ENV within the `add-ethereum-chain` template?

matthewwalsh0 replied:
This was resolved but the code appears unchanged?

NidhiKJha replied:
We don't want to render anything when we have the feature flag on. So, I think it's fine to keep it this way and instead remove it entirely when we will remove the feature flag?

matthewwalsh0 commented on ui/pages/confirmations/confirmation/templates/add-ethereum-chain.js:
```diff
@@ -245,25 +245,46 @@ function getValues(pendingApproval, t, actions, history, data) {
+            key: 'title',
+            children: originIsMetaMask
+              ? t('wantToAddThisNetwork')
+              : t('addNetworkConfirmationTitle', [
```
Could we just use a ternary on this `children` property since the rest of the element and props seem identical?

NidhiKJha replied:
Done https://github.com/MetaMask/metamask-extension/pull/25777/commits/e87feb11f778de6e03ad95819414cbb2233c4b80

matthewwalsh0 commented on ui/pages/confirmations/confirmation/templates/add-ethereum-chain.js:
```diff
@@ -245,25 +245,46 @@ function getValues(pendingApproval, t, actions, history, data) {
         element: 'Typography',
         key: 'description',
-        children: t('addEthereumChainConfirmationDescription'),
+        children: process.env.CHAIN_PERMISSIONS
```
As we reference this a few times, could we define a top level constant to avoid the duplicated `process.env`?

NidhiKJha replied:
I think keeping `process.env.CHAIN_PERMISSIONS` here should be fine for the scenario when we will be planning to remove it. Doing global search in the code base will be helpful to find where it's used and we can remove it. wdyt?

matthewwalsh0 commented on ui/pages/confirmations/confirmation/templates/add-ethereum-chain.js:
```diff
@@ -358,43 +383,84 @@ function getValues(pendingApproval, t, actions, history, data) {
-              ? pendingApproval.requestData.rpcUrl
-                  .replace(`/v3/${infuraProjectId}`, '')
+
+      process.env.CHAIN_PERMISSIONS
```
Also here, less duplication and easier to maintain if we only branch in the specific places where the values differ?

NidhiKJha replied:
Done here https://github.com/MetaMask/metamask-extension/pull/25777/commits/e87feb11f778de6e03ad95819414cbb2233c4b80

NidhiKJha committed:
updated branching

matthewwalsh0 commented on ui/components/ui/truncated-definition-list/truncated-definition-list.js:
```diff
@@ -18,57 +18,89 @@ export default function TruncatedDefinitionList({
+            warnings={warnings}
+            tooltips={tooltips}
+          />
+          <Button
```
Outside of the scope of confirmaitons, but I'd still vote to remove all possible duplication such as in this `DefinitionList` and `Button`.

NidhiKJha replied:
Sure! Done here https://github.com/MetaMask/metamask-extension/pull/25777/commits/2e86fe6f18dbf70436823dfcf9192bae1f4f2650

NidhiKJha committed:
refactored definition list

NidhiKJha committed:
updated translation

NidhiKJha committed:
updated confirmations

NidhiKJha committed:
updated conditions

NidhiKJha committed:
updated snapshot and lint errors

NidhiKJha committed:
lint fix

Commit:
Lint fix

salimtb commented:
lgtm ✅

digiwand reviewed (changes requested):
hello @NidhiKJha! The logic seems to be working well. There are some style issues I think we need to address. 

Currently, when repro'ing with:
```
await ethereum.request({
  method: 'wallet_addEthereumChain',
  params: [{
    chainId: '0x4d',
    rpcUrls: ['https://sokol.poa.network'],
    chainName: 'XDAI (Sokol)',
    nativeCurrency: { symbol: 'XDAI', decimals: 18 },
    blockExplorerUrls: ['https://blockscout.com/poa/sokol']
  }]
})
```

Without CHAIN_PERMISSIONS turned on, I am seeing a slight regression. The network container now spans across the window. 

old: 
<img width="320" src="https://github.com/user-attachments/assets/d586b2fc-f6df-4dbc-881d-6def478be087"> 

new: 
<img width="320" src="https://github.com/user-attachments/assets/4fd9df74-46f9-4cff-82c6-d03620243e91"> 

With CHAIN_PERMISSIONS turned on, I'm seeing discrepancies between Figma and the new UI 

new with CHAIN_PERMISSIONS:
<img width="320" src="https://github.com/user-attachments/assets/3ce6e39f-dd16-44f1-9b58-a05800cb3c8c">

Discrepancies observed: 
- padding around title and content should be 16px. The boxed section has margin: 24px. Above title has margin top 4px.
- description should have padding left and right 16 px 
- colors do not match for text or background
- "See details" is not centered
- footer is not sticky and is part of the scrollable content

NidhiKJha committed:
updated padding

NidhiKJha commented:
Hi @digiwand 
1. updated the padding with CHAIN_PERMISSIONS feature flag
2. Updated the network picker without feature flag
3. I couldn't see footer issue. Also, unknown networks is not a part of this PR
4. Re: See details button, we can 
![Screenshot 2024-08-27 at 6 47 17 PM](https://github.com/user-attachments/assets/e5344d6c-234f-408a-8e77-7059a5212724)
update it in a follow up PR.


![Screenshot 2024-08-27 at 7 54 01 PM](https://github.com/user-attachments/assets/4ab0f42c-7c28-46c7-88d3-6ff3f15b17b9)

digiwand reviewed (dismissed):
Nice! no blockers now. Sounds like a plan to add the See details and any other updates in a follow-up

re: padding 
There might still be excess padding between the description and rounded box. It also looks like we are missing padding to the left and right of the description 
 
 re: footer
The figma design [here](https://www.figma.com/design/BBu3ry2zLZYOmyLOEhfWYB/Multichain-v2%2B---Extension?m=dev) implies that we will want a sticky footer. In the design it shows the content is cut off while we can still see the footer

[edit: added link for figma]

NidhiKJha committed:
lint fix

jonybur reviewed (approved):
I tested this and it looks and works great. A small suggestion (non-blocker) is for us to add some margin to the text so that we don't get text all the way to the edge of the window. See image attached. This is non-blocking and/or we can change this later on in my opinion.
